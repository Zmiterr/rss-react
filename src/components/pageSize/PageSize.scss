.pageSize {
	margin-top: 15px;
}
