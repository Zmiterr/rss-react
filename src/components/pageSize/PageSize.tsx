import React, { ReactElement } from 'react';
import { useDispatch } from 'react-redux';
import { IPageSizeProps } from '../../types/types';
import { setPageSizeAct } from '../../types/actions';

import './PageSize.scss';

const PageSize = ({ pageSize }: IPageSizeProps): ReactElement => {
	const dispatch = useDispatch();

	const setPageSize = (num: number) => {
		dispatch(setPageSizeAct(num));
	};
	return (
		<div className="pageSize">
			<label htmlFor="pageSize">
				Result per page:
				<select
					className="form__select"
					name="pageSize"
					onChange={(e) => setPageSize(Number(e.target.value))}
					value={pageSize}
				>
					<option value="10">10</option>
					<option value="20">20</option>
					<option value="30">30</option>
					<option value="60">60</option>
				</select>
			</label>
		</div>
	);
};

export default PageSize;
