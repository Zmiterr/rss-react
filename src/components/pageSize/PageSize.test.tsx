import React from "react";
import PageSize from './PageSize';
//test deps
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() })

describe("PageSize component", () => {
	it("should render PageSize component", () => {
		const component = shallow(<BrowserRouter><PageSize  pageSize={50}/></BrowserRouter>);
		expect(component).toMatchSnapshot();
	});
});
