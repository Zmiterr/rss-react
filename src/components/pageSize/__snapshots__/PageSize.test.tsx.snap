// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSize component should render PageSize component 1`] = `ShallowWrapper {}`;
